File: golclinics/Arrays_and_Strings/reverseSentence.js
// Reverse Sentence In-place
const A = ['t','h','i','s',' ','i','s',' ','g','o','o','d']
function reverseSentence(A){
  helperReverse(A, 0, A.length-1);
  let startInd = 0;
  for(let i = 0; i<=A.length; i++){
    if(A[i] === ' ' || i === A.length){
      helperReverse(A, startInd, i-1);
      startInd = i+1;
    }
  }
  return A;
}

function helperReverse(array, start, end){
while(start<end){
  let temp = array[start];
  array[start] = array[end];
  array[end] = temp;
  start++;
  end--;
}
}

console.log(reverseSentence(A));

File: Yinka Olowofela/Array & Strings/ReverseSentence.py
def reverseA(sentence, start, end):

  sentence_length = end - start
  mid_sentence = (sentence_length//2)+start
  
  for i in range(start, mid_sentence):
    temp = sentence[i]
    sentence[i] = sentence[(end-1)-i+start]
    sentence[(end-1)-i+start] = temp


def reverseSentence(sentence):

  reverseA(sentence, 0, len(sentence))

  first_part = 0
  second_part = 0
  
  len_sentence = len(sentence)

  while second_part < len_sentence:
    while second_part < len_sentence and sentence[second_part] != ' ':
      second_part += 1
  
    reverseA(sentence, first_part, second_part)
    second_part += 1
    first_part = second_part

  print (sentence)
  return sentence


reverseSentence(['g','o','o','d',' ','i','s',' ', 'b','a','d'])
